src/main/java/org/folio/innreach/domain/service/impl/ContributionValidationServiceImpl.java

```diff
@@ -88,6 +89,22 @@ public boolean isEligibleForContribution(UUID centralServerId, Instance instance
       return false;
     }
 
+    var centralServer = centralServerService.getCentralServer(centralServerId);
+    var itemsEffectiveLocations = instance.getItems().stream()
+      .map(Item::getEffectiveLocationId)
+      .collect(Collectors.toList());
+    var folioLibraryIds = centralServer.getLocalAgencies().stream()
+      .map(LocalAgencyDTO::getFolioLibraryIds)
+      .collect(Collectors.toList());
+
+    var result = emptyIfNull(folioLibraryIds.stream()
+      .filter(itemsEffectiveLocations::contains)
+      .collect(Collectors.toList()));
+
+    if (result.isEmpty()) {
+      return false;
+    }
```

Comment:
This check is redundant. Each item is already being validated for contribution in this method. If no eligible items found for an instance then it is not eligible as well.

Reply:
removed